app/models/film.rb

```diff
@@ -0,0 +1,20 @@
+class Film < ActiveRecord::Base
+  belongs_to :category
+  has_many :actors, through: :film_actors
+  has_many :comments, as: :commentable
+  has_many :reviews
+  has_many :ratings
+
+  validates :name, :category_id, presence: true
+
+  def average_rating
+    @ratings = self.ratings
+    if @ratings.length > 0
```

Comment:
Use the database for this: `ratings.average(:rating)` will give you what you need